game/definitions.rpy

```diff
@@ -1590,6 +1577,20 @@ init python in jn_utils:
         """
         return list(player)[0]
 
+    def get_string_contains_profanity(string):
+        """
+        Returns True if the given string contains a profanity, based on regex.
+
+        IN:
+            - string - The string to test
+
+        OUT:
+            - True if string contains profanity; otherwise False
+        """
+        for regex in jn_globals.PROFANITY_LIST:
+            if re.search(regex, string.lower()):
+                return True
```

Comment:
Would recommend condensing the list of regex queries to one huge or chain more or less in regex. Then we can `re.compile` it which would give us some optimization on searches, also means we don't need to loop whenever doing checks